core/src/Data/DataSource/PutDataSource.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Harmony\Core\Data\DataSource;
+
+use Harmony\Core\Data\Query\Query;
+
+/**
+ * @template T
+ */
+interface PutDataSource {
+  /**
+   * @param Query $query
+   * @param T     $entity
+   *
+   * @return T
+   */
+  public function put(Query $query, mixed $entity): mixed;
```

Comment:
both query and entity should be optional

Reply:
Fixed.